vm/src/stdlib/builtins.rs

```diff
@@ -248,11 +248,35 @@ mod builtins {
     #[cfg(feature = "rustpython-compiler")]
     #[pyfunction]
     fn eval(
-        source: Either<PyStrRef, PyRef<crate::builtins::PyCode>>,
+        source: Either<ArgStrOrBytesLike, PyRef<crate::builtins::PyCode>>,
         scope: ScopeArgs,
         vm: &VirtualMachine,
     ) -> PyResult {
-        run_code(vm, source, scope, compile::Mode::Eval, "eval")
+        // source as string
+        let code = match source {
+            Either::A(either) => {
+                let source: &[u8] = &either.borrow_bytes();
+                if source.iter().any(|&b| b == 0) {
```

Comment:
```suggestion
                if source.contains(&0) {
```
I am sorry, I was forgetting about this  but slice has contains method

Reply:
I applied it in [879ed6e](https://github.com/RustPython/RustPython/pull/4052/commits/879ed6e005cb98188051b13148d9599f26974f69)